File: Data Structures & Algorithms/matrix-diagonal-sum/submission-1.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        res = 0
        m = len(mat)
        n = len(mat[0])
        for i in range(m):
            res += mat[i][i]
            res += mat[i][n-i-1]

        if n & 1:
            res -= mat[m//2][n//2]
        return res

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int j = 0;
        int res = 0;
        Set<Character> charSet = new HashSet<>();

        for(int i = 0; i<s.length(); i++) {
            while(charSet.contains(s.charAt(i))) {
                charSet.remove(s.charAt(j));
                j++;
            }
            charSet.add(s.charAt(i));
            res = Math.max(res, i-j+1);
        }
        return res;
    }
}
